downloadservice/app.py

```diff
@@ -353,3 +365,81 @@ def oauth_callback():
     next_url = auth.get_next_url(cookie_state) or url_prefix
     response = make_response(redirect(next_url))
     return response
+
+
+webdav_methods = ['GET', 'HEAD',  'MKCOL', 'OPTIONS',
+                  'PROPFIND', 'PROPPATCH', 'PUT', 'TRACE',
+                  # 'LOCK', 'UNLOCK', 'POST', 'DELETE', 'MOVE',
+                  ]
+
+
+@app.route(url_prefix + "/webdav", defaults={'path': ''},
+           methods=webdav_methods)
+@app.route(url_prefix + "/webdav/<path:path>", methods=webdav_methods)
+@authenticated
+def webdav(user, path):
+    API_HOST = urljoin_multipart(
+        app.config['CTADS_UPSTREAM_ENDPOINT'],
+        app.config['CTADS_UPSTREAM_BASEPATH'],
+    )
+
+    def is_chunked():
+        return request.headers.get('transfer-encoding', '').lower() \
+            == 'chunked'
+
+    def request_datastream():
+        while (buf := request.stream.read(default_chunk_size)) != b'':
+            yield buf
+
+    if request.method in ['PUT', 'MKCOL', 'PROPPATCH']:
```

Comment:
thinking more I am concerned about possible future issues like forgetting `PROPPATCH`, maybe it's best to forbid everything except for an explicit list
```suggestion
    if request.method not in ['GET', 'HEAD', 'OPTIONS', 'PROPFIND', 'TRACE']:
```suggestion
    if request.method in ['PUT', 'MKCOL', 'PROPPATCH']:
```
```suggestion
    if request.method in ['PUT', 'MKCOL', 'PROPPATCH']:
```
```

Reply:
I realize it is correctly handled at this moment, my point is that with my suggestion a mistake is less likely to be introduced with some change.

Reply:
for example, if one uncomments `DELETE` because we decide to support it, we have to remeber to add it here. It's safer not to introduce this sort of dependency, it's an opportunity for a future mistake.

Reply:
I fixed it in between your comments, I totally agree. 👍